### DISCUSSION TITLE
Strange behavior at first startup

### DISCUSSION THREAD

**Discussion by Akkato47** [mikro-orm/mikro-orm#6043] [General]
### Describe the bug

I'm not quite sure how to describe it, but here's the problem:
The first time I run the program, when the database has just been populated I have a problem getting data from new users. The problem is that if a user is created and almost immediately makes a request for full information about himself, the response comes absolutely all fields, including the password, etc., although I specify both in exclude and fields which fields I need, and which are not, also there are strange duplicates. After restarting the application everything comes back to normal, but if you create a new record, everything repeats and again passes only after restarting the application.

### Reproduction

Here is the code:

```
const data = await DI.em.findOne(
            Users,
            { tag },
            {
                populate: [
                    "profileInfo.userEducation",
                    "profileInfo.userExperience",
                    "profileInfo.userInterests",
                    "profileInfo.userLocation",
                    "profileInfo.userSkills",
                    "profileInfo.files",
                ],
                exclude: ["password"],
            },
        );
```

And here is the first answer:
```
{
  "uid": "daccf825-dcbe-47d2-a94b-4fa6ff50baf0",
  "utr": [],
  "fullName": "string",
  "tag": "EINKPPUZ",
  "mail": "string",
  "password": "$2b$10$YQM43mBEFXMU4Si2Es64MuMW/GP9nVpzTDBa9kO8RW.j8YmTaln6C",
  "phone": "string",
  "role": "ORG",
  "birthDate": "2024-09-18T00:00:00.000Z",
  "via": "BASE",
  "profileInfo": {
    "uid": "3b653014-93d1-498d-9b2f-7e857679d4d2",
    "userExperience": [],
    "userInterests": [],
    "userSkills": [],
    "files": [],
    "isSearchingJob": false,
    "status": "",
    "about": "",
    "user": {
      "uid": "daccf825-dcbe-47d2-a94b-4fa6ff50baf0",
      "utr": [],
      "fullName": "string",
      "tag": "EINKPPUZ",
      "mail": "string",
      "password": "$2b$10$YQM43mBEFXMU4Si2Es64MuMW/GP9nVpzTDBa9kO8RW.j8YmTaln6C",
      "phone": "string",
      "role": "ORG",
      "birthDate": "2024-09-18T00:00:00.000Z",
      "via": "BASE"
    }
  }
}
```

Here is the answer after app restart:
```
{
  "uid": "daccf825-dcbe-47d2-a94b-4fa6ff50baf0",
  "fullName": "string",
  "tag": "EINKPPUZ",
  "mail": "string",
  "phone": "string",
  "role": "ORG",
  "birthDate": "2024-09-18T00:00:00.000Z",
  "image": null,
  "backgroundImage": null,
  "via": "BASE",
  "profileInfo": {
    "uid": "3b653014-93d1-498d-9b2f-7e857679d4d2",
    "isSearchingJob": false,
    "status": "",
    "about": "",
    "user": "daccf825-dcbe-47d2-a94b-4fa6ff50baf0",
    "userEducation": null,
    "userLocation": null,
    "userExperience": [],
    "userInterests": [],
    "userSkills": [],
    "files": []
  }
}
```

My mikro-orm config:
```
const options: Options<PostgreSqlDriver> = {
    driver: PostgreSqlDriver,
    entities: ["./dist/src/db/mikro-orm/entities/**/*.entity.js"],
    entitiesTs: ["./src/db/mikro-orm/entities/**/*.entity.ts"],
    dbName: config.database.postgres.database,
    password: config.database.postgres.password,
    user: config.database.postgres.user,
    host: config.database.postgres.host,
    port: parseInt(config.database.postgres.port || "5432"),
    migrations: {
        tableName: "mikro_orm_migrations",
        allOrNothing: true,
        path: path.join(process.cwd(), "dist/src/db/mikro-orm/migrations"),
        pathTs: path.join(process.cwd(), "src/db/mikro-orm/migrations"),
    },
    allowGlobalContext: true,
    forceUtcTimezone: true,
    timezone: "+03:00",
    debug: true,
    extensions: [SeedManager],
    seeder: {
        path: "./dist/src/db/mikro-orm/seeders",
        pathTs: "./src/db/mikro-orm/seeders",
        defaultSeeder: "DatabaseSeeder",
        glob: "!(*.d).{js,ts}",
        emit: "ts",
    },
    serialization: {
        forceObject: false,
        includePrimaryKeys: false,
    },
    persistOnCreate: true,
}
```

### What driver are you using?

@mikro-orm/postgresql

### MikroORM version

6.3.7

### Node.js version

20.11.1

### Operating system

Windows

### Validations

- [X] Read the [Contributing Guidelines](https://github.com/mikro-orm/mikro-orm/blob/master/CONTRIBUTING.md).
- [X] Read the [docs](https://mikro-orm.io/docs).
- [X] Check that there isn't [already an issue](https://github.com/mikro-orm/mikro-orm/issues) that reports the same bug to avoid creating a duplicate.
- [X] Check that this is a concrete bug. For Q&A open a [GitHub Discussion](https://github.com/mikro-orm/mikro-orm/discussions) or join our [Discord](https://discord.gg/w8bjxFHS7X).
- [x] The provided reproduction is a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

**Comment by B4nan** [comment]
> Here is the code:

This is not a reproduction.

Your problem is most likely that the find call returns the entity from identity map - its not making any database query, it just gives you what you have in the memory. And that record indeed has password set.

Using `exclude` for this feels wrong to me, or should I say error prone. Better to make the property `lazy: true` (which means it wont be selected unless explicitly populated) and `hidden: true` so it never leaks during serialization. That way you don't have to care about this.

Alternative would be storing the user in a separate context (`em.fork()` in the place where you create it) or using `refresh: true` when you read from the database (so you replace the record).
